src/llama_stack_client/resources/inference.py

```diff
@@ -930,6 +936,8 @@ async def completion(
         extra_body: Body | None = None,
         timeout: float | httpx.Timeout | None | NotGiven = NOT_GIVEN,
     ) -> CompletionResponse | AsyncStream[CompletionResponse]:
+        if stream:
```

Comment:
why only inference? does agent also need the headers?

Reply:
The header is required by sherpa server only, since it ignores stream body param due to some limitations. I would postpone changing others until we get there.